packages/model-viewer/src/features/scene-graph/api.ts

```diff
@@ -39,7 +43,29 @@ export declare interface Model {
    * correspond to the listing of materials in the glTF, with the possible
    * addition of a default material at the end.
    */
-  readonly materials: Readonly<Array<Material>>;
+  readonly materials: Readonly<Material[]>;
+
+  /**
+   * Gets a material(s) by name.
+   * @param name the name of the material to return.
+   * @returns a list of materials who's name is equal to `name` or an empty list
+   *     if no matching materials found.
+   */
+  getMaterialByName(name: string): Material[];
+
+  /**
+   * Intersects a ray with the Model and returns a list of materials who's
+   * objects were intersected.
+   * @param screenX 2D X coordinates of the mouse, in normalized device
+   *     coordinates (NDC)---X component should be between -1 and 1.
+   * @param screenX 2D Y coordinates of the mouse, in normalized device
+   *     coordinates (NDC)---Y component should be between -1 and 1.
+   * @param camera camera from which the ray should originate
+   * @returns a list of materials, if no intersection is made the an empty list
+   *     is returned.
+   */
+  intersectMaterial(screenX: number, screenY: number, camera: Camera):
```

Comment:
Why do we take a camera as input?

Reply:
We discussed this, but intersect methods are typically low level and don't make any assumptions about where a camera comes from, updates in coming based on in person feedback.